src/server.cpp

```diff
@@ -142,6 +144,8 @@ void *ServerThread::run()
 			m_server->setAsyncFatalError(e.what());
 		} catch (LuaError &e) {
 			m_server->setAsyncFatalError(e);
+		} catch (ModError &e) {
+			m_server->setAsyncFatalError(e.what());
```

Comment:
are we using `ModError` somewhere when we should be using `LuaError` or what's the root cause?

Reply:
Because https://github.com/minetest/minetest/pull/13987#discussion_r1703340146
It's a logic issue that is optimally checked in the setter function.